src/lib/core/fiber.c

```diff
@@ -1077,9 +1077,11 @@ fiber_loop(MAYBE_UNUSED void *data)
 }
 
 void
-fiber_set_name(struct fiber *fiber, const char *name)
+fiber_set_name_n(struct fiber *fiber, const char *name, uint32_t len)
 {
-	size_t size = strlen(name) + 1;
+	if (fiber == NULL)
+		fiber = fiber();
```

Comment:
I wouldn't do it for some of the functions, it makes the API inconsistent. All the other public API `fiber_*()` functions that accept `struct fiber *` doesn't assume that `NULL` means the current fiber.

Reply:
I would really appreciate, if they did. Making a `fiber_self` ffi call each time is both less efficient (impossible to inline) and less ergonomic for the programmer.

What do you think about adding the same functionality to other fiber_* functions?